moo.js

```diff
@@ -240,7 +251,10 @@
     var match = this.eat(re)
     var group, value, text
     if (match === null) {
-      group = Lexer.error
+      group = this.error
+      if (!group) {
+        throw new Error('Syntax error')
```

Comment:
This is fine for now, but perhaps we should think about making it easy for us/users to generate nice syntax errors—with line/column information and a preview of the buffer pointing to the problematic character, for example.